v0/destinations/facebook_pixel/transform.js

```diff
@@ -491,6 +492,10 @@ const responseBuilderSimple = (
           commonData.event_name = "Purchase";
           break;
         case "products searched":
+          const query = message.properties?.query;
+          if (query && typeof query !== "string" && typeof query !== "number") {
+            throw new CustomError("'query' should be in (string (or) number) format only", 400);
```

Comment:
According to Rudderstack spec, a Query should be a string or object. So an error message like `string or number`, would that not be misleading? WDYT @ItsSudip @utsabc

Reply:
Yeah, that is correct. But we are throwing this error as Facebook doesn't support `object`.

Reply:
In that case comment there with your findings. 
like: 
```
/* custom_data field type ref : < link to the page>
    and also by behaviour it is found that it supports `boolean` and `number` too. As a result these two types of fields are allowed */
```

Reply:
Done